zc_install/includes/nginx_conf/ngx_conf_server.txt

```diff
@@ -0,0 +1,320 @@
+### 		                  NGINX CONFIGURATION FOR ZEN CART
+###
+### NOTES: 
+###    - Copy and paste into ‘server’ section of nginx conf file for ZEN CART
+###    - Location Block order is important
+###    - Paste before any other location blocks if any
+###
+###
+####################################################################################################
+####################################################################################################
+####################################################################################################
+	
+	
+	#############################################################################################
+	# ZEN CART PHP Handler
+	#############################################################################################	
+	## The following block serves all PHP requests internally directed to it within this conf file
+	## Edit to reflect your actual setup for serving PHP
+	location @zencart_php_handler {
+		if (!-f $document_root$fastcgi_script_name) { return 404; }
+
+		fastcgi_index	index.php;
+		fastcgi_pass	12.0.0.1:9000;
+		
+		fastcgi_param	SCRIPT_FILENAME		$document_root$fastcgi_script_name;
+		fastcgi_param	QUERY_STRING		$query_string;
+		fastcgi_param	REQUEST_METHOD		$request_method;
+		fastcgi_param	CONTENT_TYPE		$content_type;
+		fastcgi_param	CONTENT_LENGTH		$content_length;
+		fastcgi_param	SCRIPT_NAME			$fastcgi_script_name;
+		fastcgi_param	REQUEST_URI			$request_uri;
+		fastcgi_param	DOCUMENT_URI		$document_uri;
+		fastcgi_param	DOCUMENT_ROOT		$document_root;
+		fastcgi_param	SERVER_PROTOCOL		$server_protocol;
+		fastcgi_param	GATEWAY_INTERFACE	CGI/1.1;
+		fastcgi_param	SERVER_SOFTWARE		nginx/$nginx_version;
+		fastcgi_param	REMOTE_ADDR			$remote_addr;
+		fastcgi_param	REMOTE_PORT			$remote_port;
+		fastcgi_param	SERVER_ADDR			$server_addr;
+		fastcgi_param	SERVER_PORT			$server_port;
+		fastcgi_param	SERVER_NAME			$server_name;
+		fastcgi_param	REDIRECT_STATUS		200;
+		
+		fastcgi_hide_header X-Powered-By;
+	}
+	
+	
+	#############################################################################################
+	# ZEN CART Pretty URLs Handler
+	#############################################################################################	
+	## The following block handles Pretty URLs
+	## Insert directives below as required
+	#
+	#
+	#
+	
+	
+	#############################################################################################
+	# ZEN CART Root Folder
+	#############################################################################################	
+	## Allow all requests to specific static file types
+	location ~ %%store_folder%%/.+\.(md|ico|txt|html)$ {
+		expires $zencart_expires;
+	}
+	## Allow all requests to specific PHP files directly under the ZEN CART ROOT folder
+	location ~ %%store_folder%%/(ajax|index|page_not_found|ipn_main_handler)\.php$ {
+		error_page 418 = @zencart_php_handler;
+		return 418;
+	}
```

Comment:
While I appreciate the easter egg "coffee/teapot" 418 humour, isn't it better practice to actually server a 200 response when in fact there is no error? `4xx` codes are errors. There's no error here.

Or with nginx is the use of 418 becoming a "best practice" despite the semantic confusion between purposes of 2xx and 4xx codes?